packages/gatsby-plugin-page-creator/src/__tests__/gatsby-node.js

```diff
@@ -18,6 +18,59 @@ describe(`JavaScript page creator`, () => {
     expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
   })
 
+  it(`filters out files that start with dot`, () => {
+    const files = [
+      {
+        path: `.eslintrc`,
+      },
+      {
+        path: `something/.eslintrc`,
+      },
+      {
+        path: `something/.eslintrc.js`,
+      },
+      {
+        path: `something/blah.js`,
+      },
+      {
+        path: `.markdownlint.json`,
+      },
+      {
+        path: `something/.markdownlint.json`,
+      },
+    ]
+
+    expect(files.filter(file => validatePath(file.path)).length).toEqual(1)
```

Comment:
nit: but I'd probably just remove the `something/blah.js` and validate that the array length is `0`. _Then_ test the explicit .js inclusion in a separate test?